### PR TITLE
feat(config)!: merge `timeout_prevote` and `timeout_precommit`

### LINKED ISSUE #2893: Merge `timeout_prevote_delta` & `timeout_precommit_delta` into a single `timeout_vote_delta`
`timeout_prevote` & `timeout_precommit` into a single `timeout_vote`

### PR DESCRIPTION
**Pull request by melekes**
`timeout_prevote_delta` and `timeout_precommit_delta`

into `timeout_vote` and `timeout_vote_delta` accordingly

Closes #2893

---

#### PR checklist

- [ ] ~~Tests written/updated~~
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: config/config.go
Comment (sergio-mena) [inline config/config.go:1106]: `TimeoutVote`?

File: docs/references/config/config.toml.md
Comment (sergio-mena) [inline docs/references/config/config.toml.md:1623]: We need a good way to minimize disruption for operators upgrading from current config (and not thinking of adapting this, so the new field will have value 0).
An idea can be that the new timeout, when 0, means some sensible default value (e.g. our default value: 1s).
Anyway, disabling these timeouts never made any sense to me in production. And if need to disable it in text, we can just set it to `time.Hour`.

Also, this design would enable us to (later on) add those timeouts as ConsensusParams:
* If the value is 0, the node uses the consensus param value
* But if the operator wants to override the consensus param value, they can by setting a value != 0 in the `config.toml`

Let me know what you think

Comment (melekes) [reply docs/references/config/config.toml.md]: While this would work in this case, it won't in the case of `skip_timeout_commit`. If the user had `skip_timeout_commit=true` and `timeout_commit=1s`, in v1 if he/she doesn't do anything, the node will be running with `1s` timeout commit.

What if instead we version config and write a CLI tool for upgrading it? If the manual cmd is not desirable, then we can attempt upgrade upon a startup. How does that sound?

Comment (sergio-mena) [reply docs/references/config/config.toml.md]: Fair enough, but it risks being quite some effort. Maybe, a middle ground would be:
* We version the config as you say
* We include all changes to the config in a paragraph in `UPGRADING.md`
* Upon start-up, if the version received is
  * In the future, we panic saying this is an old version of CometBFT
  * In the past, we warn the user that they may have some fields that became obsolete
    * we include a link to the paragraph in `UPGRADING.md` to explain what to change
    * alternatively, for safely, we panic and ask the user to
      * bump the version manually
      * while doing that, to pay attention at the config fields that changed

If we decide to go the config tool path (I think it's overkill for the changes we have ATM), there is already one, called Confix, developed by the previous Tendermint Core team, so we should probably reuse it. It's on `v0.36.x` and `master`. AFAICT the SDK folks are also using it.

Comment (sergio-mena) [reply docs/references/config/config.toml.md]: Anyway, this discussion should block this PR (although the value 0 to `timeout_round` when upgrading isn't solved)

**Review by sergio-mena — APPROVED** [review]

File: docs/references/config/config.toml.md
Comment (sergio-mena) [reply docs/references/config/config.toml.md]: @melekes FYI, I'm adding a point in the community call, to see what users think

Comment (melekes) [reply docs/references/config/config.toml.md]: I will address this in a separate PR.

**Pull request merged**
